pkg/apiserver/slowquery/queries.go

```diff
@@ -106,46 +101,61 @@ type GetListRequest struct {
 	// for showing slow queries in the statement detail page
 	Plans  []string `json:"plans" form:"plans"`
 	Digest string   `json:"digest" form:"digest"`
-}
 
-type GetDetailRequest struct {
-	Digest    string  `json:"digest" form:"digest"`
-	Time      float64 `json:"time" form:"time"`
-	ConnectID int64   `json:"connect_id" form:"connect_id"`
+	Fields string `json:"fields" form:"fields"` // example: "Query,Digest"
 }
 
-func getAllColumnNames() []string {
-	t := reflect.TypeOf(Base{})
+func getRefColumns(jsonFields ...string) ([]string, error) {
+	fields := make(map[string]*reflect.StructField)
+	t := reflect.TypeOf(SlowQuery{})
 	fieldsNum := t.NumField()
-	ret := make([]string, 0, fieldsNum)
 	for i := 0; i < fieldsNum; i++ {
 		field := t.Field(i)
-		if s, ok := field.Tag.Lookup("gorm"); ok {
-			list := strings.Split(s, ":")
-			if len(list) < 1 {
-				panic(fmt.Sprintf("Unknown gorm tag field: %s", s))
+		fields[strings.ToLower(field.Tag.Get("json"))] = &field
+	}
+	ret := make([]string, 0, len(jsonFields))
+	for _, fieldName := range jsonFields {
+		if field, ok := fields[strings.ToLower(fieldName)]; ok {
+			if proj, ok := field.Tag.Lookup("proj"); ok {
+				ret = append(ret, fmt.Sprintf("%s AS %s", proj, gorm.ToColumnName(field.Name)))
```

Comment:
`gorm.ToColumnName(field.Name)` might be incorrect. What you want may be `field.Tag.Lookup()` instead?